esmvalcore/preprocessor/_multimodel.py

```diff
@@ -260,13 +256,103 @@ def _equalise_coordinates(cubes):
                 break
         else:
             equal_coords_metadata.append(coord.metadata)
+    return equal_coords_metadata
+
+
+def _get_equal_coord_names_metadata(cubes, equal_coords_metadata):
+    """Get metadata for coords with matching names and units across cubes.
+
+    Note
+    ----
+    Ignore coordinates whose names are not unique.
+
+    """
+    equal_names_metadata = {}
+    for coord in cubes[0].coords():
+        coord_name = coord.name()
+
+        # Ignore exactly matching coordinates
+        if coord.metadata in equal_coords_metadata:
+            continue
+
+        # Ignore coordinates that are not unique in original cube
+        if len(cubes[0].coords(coord_name)) > 1:
+            continue
+
+        # Check if coordinate names and units match across all cubes
+        for other_cube in cubes[1:]:
+
+            # Ignore names that do not exist in other cube/are not unique
+            if len(other_cube.coords(coord_name)) != 1:
+                break
+
+            # Ignore names where units do not match across cubes
+            if coord.units != other_cube.coord(coord_name).units:
+                break
 
-    # Modify coordinates accordingly
+        # Coordinate name exists in all other cubes with identical units
+        # --> Get metadata that is identical across all cubes
+        else:
+            std_names = list(
+                {c.coord(coord_name).standard_name for c in cubes}
+            )
+            long_names = list(
+                {c.coord(coord_name).long_name for c in cubes}
+            )
+            var_names = list(
+                {c.coord(coord_name).var_name for c in cubes}
+            )
+            equal_names_metadata[coord_name] = dict(
+                standard_name=std_names[0] if len(std_names) == 1 else None,
+                long_name=long_names[0] if len(long_names) == 1 else None,
```

Comment:
what if `long_name = ["some_long_name", None]` since all datasets have long name set to `some_long_name` bar one annoying orange that's `None`, you gonna set them all to `None` only if one is `None`?

Reply:
Yeah, they all would be set to `None`. This is a really difficult question. Where would you draw the line? If 2/10 are `None`? 3/10? 9/10? I think the current solution is the most straightforward one.

Reply:
yeah it is, maybe we should think about this - in a different PR - this is up for merging now, you wanna invoke the Gods of the Tech group for one of them to merge it?

Reply:
BTW I think we should apply the same logic behind the old saying "if ten guys tell you you're drunk, you should go home" and think of a decent statistic, maybe 95% of them - 95CI? :grin:

Reply:
Another possibility would be to concatenate all non-None names and to allow for an (optional) preprocessor parameter. But let's leave that for the future.

As an aside, if you want to avoid this kind of comment in the post-review phase, resolve the conversations when you are done with them :wink: